src/themes/material/components/edit-chip-list/edit-chip-list.component.ts

```diff
@@ -0,0 +1,198 @@
+import {
+	Component,
+	Input,
+	OnInit,
+	ChangeDetectorRef,
+	ViewChildren,
+	QueryList
+} from '@angular/core';
+
+import {AppError} from 'ng2-qgrid/core/infrastructure';
+import {Command, CommandManager} from 'ng2-qgrid/core/command';
+import {Shortcut, ShortcutManager} from 'ng2-qgrid/core/shortcut';
+import {EditChipComponent} from './edit-chip/edit-chip.component';
+import {ChipState} from './edit-chip/edit-chip.component';
+
+interface ISelected {
+	index?: number;
```

Comment:
why u need this abstraction?

Reply:
This abstraction is represented a selected chip object, that needs to store a few property values. I think it is a better option rather than store the properties of a single object (obviously, it is an object by its nature) in a few independent variables and manipulated with them in a designated function in one place like
 ` updateSelectedValues(index, value, action) `
 (although this is an option as well). This would be more like procedural programming and seems to make the code harder to understand. But maybe I am wrong?

Reply:
i'm not sure if u need `updateSelectedValues(index, value, action)` i agree that should be selectionIndex property all other state looks redundant from mine point of view.

Reply:
I've removed this abstraction after refactoring in the commit: [Merge from Dev: conflicts resolving](https://github.com/qgrid/ng2/pull/39/commits/da05f15a228a611a5059e3049c9f18402446fa8f).
Moreover, since the IChipItem abstraction still remains there was no need to use selectionIndex property. So it wasn't added.